components/core/src/compressor_frontend/Buffer.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef COMPRESSOR_FRONTEND_BUFFER_HPP
+#define COMPRESSOR_FRONTEND_BUFFER_HPP
+
+// C++ libraries
+#include <cstdint>
+#include <vector>
+
+// Project Headers
+#include "Constants.hpp"
+
+/**
+ * A base class for keeping track of static and dynamic buffers needed for a growing buffer.
+ * The base class does not grow the buffer, the child class is responsible for doing this.
+ */
+
+namespace compressor_frontend {
+    template <typename type>
+    class Buffer {
+    public:
+        // Prevent copying of buffer as this will be really slow
+        Buffer (Buffer&&) = delete;
```

Comment:
* If you're deleting the move constructor, you should delete the copy constructor as well ([see rule of 5](https://cpppatterns.com/patterns/rule-of-five.html)).
* Why're you deleting the move constructor?
* The move constructor should come after the constructor.